minesweeper.py

```diff
@@ -85,26 +85,40 @@ def open_squares(self, x, y):
         if coordinate in self.previously_clicked_square or coordinate in self.flag_dict:
             return
 
-        self.previously_clicked_square.add((x, y))
-        canvas.create_image(
-            int(x * button_size),
-            int(y * button_size),
-            image=button_image_pressed,
-            anchor="nw",
-        )
+        if coordinate in self.mine_locations:
+            self.previously_clicked_square.add((x, y))
+            canvas.create_image(
+                int(x * button_size),
+                int(y * button_size),
+                image=red_button_image_pressed,
+                anchor="nw",
+            )
+        else:
+            self.previously_clicked_square.add((x, y))
```

Comment:
Please move `self.previously_clicked_square.add((x, y))` before `if`,`else` because it's the same in both cases.

Alternatively, you can use [a conditional expression](https://stackoverflow.com/questions/394809/does-python-have-a-ternary-conditional-operator):

```
            canvas.create_image(
                ...,
                image=(red_button_image_pressed if coordinate in self.mine_locations else button_image_pressed),
                ...,
            )
```